From f82cec5ae917e4d43a2a003b2eb0ddae3c0d31c7 Mon Sep 17 00:00:00 2001
From: Brian Heim <brianlheim@gmail.com>
Date: Sun, 4 Mar 2018 18:29:54 -0500
Subject: [PATCH 1/2] cmake: correctly include yaml-cpp

Include its dirs last for libsclang, and use the YAMLCPP_LIBRARY variable for linking
---
 editors/sc-ide/CMakeLists.txt | 3 ++-
 lang/CMakeLists.txt           | 4 ++--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/editors/sc-ide/CMakeLists.txt b/editors/sc-ide/CMakeLists.txt
index ffe70f98e..2f6c9c2cc 100644
--- a/editors/sc-ide/CMakeLists.txt
+++ b/editors/sc-ide/CMakeLists.txt
@@ -199,6 +199,7 @@ include_directories(${CMAKE_SOURCE_DIR}/include/common)
 include_directories(${CMAKE_SOURCE_DIR}/common)
 include_directories(${CMAKE_SOURCE_DIR}/include/plugin_interface)
 include_directories(${CMAKE_SOURCE_DIR}/external_libraries/boost)
+include_directories(${YAMLCPP_INCLUDE_DIR})
 # For QtCollider headers:
 include_directories(${CMAKE_SOURCE_DIR})
 # Needed for auto-generated forms headers:
@@ -273,7 +274,7 @@ endif()
 
 target_link_libraries( SuperCollider
     ${QT_IDE_LIBRARIES}
-    yaml
+    ${YAMLCPP_LIBRARY}
     oscpack
 )
 
diff --git a/lang/CMakeLists.txt b/lang/CMakeLists.txt
index 9270b6320..5de19e46e 100644
--- a/lang/CMakeLists.txt
+++ b/lang/CMakeLists.txt
@@ -10,8 +10,6 @@ include_directories(${CMAKE_SOURCE_DIR}/include/common
                     ${CMAKE_SOURCE_DIR}/include/server
                     ${CMAKE_SOURCE_DIR}/common
 
-                    ${YAMLCPP_INCLUDE_DIR}
-
                     ${CMAKE_SOURCE_DIR}/external_libraries/boost_sync/include
 
                     LangSource
@@ -19,6 +17,8 @@ include_directories(${CMAKE_SOURCE_DIR}/include/common
 
                     ${CMAKE_SOURCE_DIR}/external_libraries/nova-tt
                     LangSource/Bison
+
+                    ${YAMLCPP_INCLUDE_DIR}
 )
 
 
-- 
2.16.2

